lib/logidze/model.rb

```diff
@@ -94,6 +94,14 @@ def at(time: nil, version: nil)
 
       build_dup(log_entry, time)
     end
+
+    def logidze_versions(reverse: false, include_self: false)
+      versions_meta = reverse ? log_data.versions.reverse : log_data.versions
+      versions_meta = versions_meta.reject { _1.version == log_data.version } unless include_self
+
+      versions_meta.map { at(version: _1.version) }.to_enum
```

Comment:
We wanted this method to return an enumerator to avoid expensive calculation and object creation of generating all versions, i.e., when I call `record.logidze_versions.take(2)` I only call `#at` twice. Right how, we create an array of all version first—this is not optimal.

We can add a test to check that we don't create a record per version by adding an `after_initialize` callback and increment some counter within it.